refactored tests

orthagonal opened the pull request:
refactored tests (one file containing server/network tests and the other
testing various packet formats)
added additional test cases
exposed makeSlackPayload as a server method

orthagonal committed:
refactored tests

refactored tests (one file containing server/network tests and the other
testing various packet formats)
added additional test cases
exposed makeSlackPayload as a server method

jgallen23 commented on test/package-test.js:
```diff
@@ -0,0 +1,172 @@
+    const packetString = server.makeSlackPayload(['test'], 'a string');
+    code.expect(typeof packetString).to.equal('string');
+    const packet = JSON.parse(packetString);
+    code.expect(packet.attachments.length).to.equal(1);
```
probably easier to do a deep equal and just put the whole attachment in there

jgallen23 replied:
lets switch everything to deep equal so we are making sure there aren't any other things getting added to payload that we weren't expecting

jgallen23 commented on test/package-test.js:
```diff
@@ -0,0 +1,172 @@
+  });
+});
+lab.test('lets you post an object as the message', (done) => {
+  server = new Hapi.Server({});
```
you could move this into setup. so each test is just calling server.makeSlackPayload

jgallen23 replied:
nevermind, I see why you did that.

jgallen23 commented on test/package-test.js:
```diff
@@ -0,0 +1,172 @@
+});
+lab.test('converts a basic message passed as string ', (done) => {
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
```
is this required? would be great to have potential port conflicts when spinning up tests

jgallen23 commented on test/package-test.js:
```diff
@@ -0,0 +1,172 @@
+    options: {}
+  }, () => {
+    const packet = JSON.parse(server.makeSlackPayload([], {
+      message: 'this is the message that was pulled out of the object below',
```
need to add another test for `url` too.  will set the title_link field

jgallen23 replied:
need a test to test the object if it doesn't have a message field

jgallen23 replied:
need a test to test adding the fallback field (will get covered if you do a deep equal on the object)

orthagonal replied:
Can you clarify?  I don't think we have a fallback option currently.

jgallen23 replied:
make sure you merge in from master

jgallen23 commented on test/package-test.js:
```diff
@@ -0,0 +1,172 @@
+    done();
+  });
+});
+lab.test('will put a link in the title', (done) => {
```
I think this is mis-labeled

jgallen23 commented:
can we turn on code coverage? would it be at 100%?

orthagonal committed:
tests updated

changed tests to use 'deepEqual' instead of comparing individual fields
in the packet
fixed bug that was causing data.url to end up in the packet
turned on code coverage (currently at 95)

orthagonal committed:
fixed mis-labelled unit test

also removed "port: 8080" from packet tests

jgallen23 commented on test/package-test.js:
```diff
@@ -0,0 +1,325 @@
+    const packetString = server.makeSlackPayload(['test'], 'a string');
+    code.expect(typeof packetString).to.equal('string');
+    const packet = JSON.parse(packetString);
+    code.expect(_.isEqual(packet, expectedPacket)).to.equal(true);
```
https://github.com/hapijs/code/blob/master/API.md#equalvalue-options

orthagonal replied:
Nice, I didn't know code had this built in already.

jgallen23 commented on test/package-test.js:
```diff
@@ -0,0 +1,325 @@
+      mrkdwn_in: ['text'],
+      fields: [{
+        title: 'Tags',
+        value: ''
```
we should skip the tags field if no tags are passed

orthagonal committed:
updated code version

updated code version to 3.0.0 (by default does deep equal) and changed
tests to match
does not add 'Tags' field if no tags are present

orthagonal committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into better-output

orthagonal committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into better-output

orthagonal committed:
added 'fallback' to tests
